### PR TITLE
.github: Migrate iOS workflows to GHA

### LINKED ISSUE #67643: Migrate iOS workflows to GHA
We'd like to migrate the current iOS configurations to GHA:

- pytorch_ios_12_5_1_arm64_build
- pytorch_ios_12_5_1_arm64_coreml_build
- pytorch_ios_12_5_1_arm64_custom_build
- pytorch_ios_12_5_1_arm64_full_jit_build
- pytorch_ios_12_5_1_arm64_metal_build
- pytorch_ios_12_5_1_x86_64_build
- pytorch_ios_12_5_1_x86_64_coreml_build
- pytorch_ios_12_5_1_x86_64_full_jit_build

Current definitions exist here:
https://github.com/pytorch/pytorch/blob/45d5b3248b430aca70111316accd165954464589/.circleci/cimodel/data/simple/ios_definitions.py#L63-L84

Since the iOS builds rely on a root cert being available (and the only sane way we have of supplying that root cert is through github secrets), I'm going to suggest that we limit the builds to only master and only to PRs that have the `ciflow/ios` label

cc @seemethere @malfet @pytorch/pytorch-dev-infra

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#67645 .github: Migrate iOS workflows to GHA**

This PR migrates the iOS workflows to GHA from CircleCI. Also makes it so that iOS builds will run when `ciflow/ios` is added to a PR and will by default only run on `master`

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D32104367](https://our.internmc.facebook.com/intern/diff/D32104367)

### CODE REVIEW

**Review by janeyx99 — COMMENTED** [review]
It seems that these three are run on PRs currently--are they now moved to trunk only?
image

**Review by malfet — APPROVED** [review]

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32104367).

File: tools/lite_interpreter/gen_selected_mobile_ops_header.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def main() -> None:
     print("Loading yaml file: ", model_file_name)
     loaded_model = {}
     with open(model_file_name, "rb") as model_file:
-        loaded_model = yaml.load(model_file)
+        loaded_model = yaml.safe_load(model_file)
```
Comment (seemethere) [inline tools/lite_interpreter/gen_selected_mobile_ops_header.py:130]: Changed to `safe_load` since `load` now leads a `Loader` argument

**Comment by seemethere** [comment]
Currently waiting on https://github.com/pytorch/pytorch/pull/67694, will rebase after that gets merged

**Comment by seemethere** [comment]
> It seems that these three are run on PRs currently--are they now moved to trunk only? image

Yes considering that the iOS builds don't fail often, rely on expensive / slow executors (macOS), and are specialized in general I think it's pretty safe to move them to trunk only and let iOS PyTorch devs set the default ciflow for them to include `ciflow/ios`

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32104367).

**Pull request closed without merging**